diff -Naur lfs-bootscripts-20190908-ori/lfs/init.d/mountvirtfs lfs-bootscripts-20190908/lfs/init.d/mountvirtfs
--- lfs-bootscripts-20190908-ori/lfs/init.d/mountvirtfs	2019-09-02 16:53:46.000000000 +0800
+++ lfs-bootscripts-20190908/lfs/init.d/mountvirtfs	2019-10-10 23:43:59.889964616 +0800
@@ -56,6 +56,11 @@
          mount -o mode=0755,nosuid /dev  || failed=1
       fi
 
+      if [ -d /sys/firmware/efi/efivars ]; then
+         log_info_msg2 " ${INFO}efivarfs" 
+         mount -t efivarfs efivarfs /sys/firmware/efi/efivars
+      fi
+
       ln -sfn /run/shm /dev/shm
       
       (exit ${failed})
