diff -Naur pkgutils-5.40.7-ori/pkgmk.in pkgutils-5.40.7/pkgmk.in
--- pkgutils-5.40.7-ori/pkgmk.in	2019-09-25 16:43:09.157258000 +0800
+++ pkgutils-5.40.7/pkgmk.in	2019-09-25 16:46:06.927313600 +0800
@@ -586,9 +586,9 @@
 	info "Installing '$TARGET'."
 
 	if [ "$PKGMK_INSTALL" = "install" ]; then
-		COMMAND="pkgadd $TARGET"
+		COMMAND="pkgadd $PKGMK_IGNORE_CONFLICT $TARGET"
 	else
-		COMMAND="pkgadd -u $TARGET"
+		COMMAND="pkgadd -u $PKGMK_IGNORE_CONFLICT $TARGET"
 	fi
 
 	cd $PKGMK_ROOT
@@ -684,6 +684,7 @@
 	echo "  -u,   --upgrade             build and install package (as upgrade)"
 	echo "  -r,   --recursive           search for and build packages recursively"
 	echo "  -d,   --download            download missing source file(s)"
+	echo "  -ic,  --ignore-conflict     ignore conflicting files when install/upgrade"
 	echo "  -do,  --download-only       do not build, only download missing source file(s)"
 	echo "  -eo,  --extract-only        do not build, only extract source file(s)"
 	echo "  -utd, --up-to-date          do not build, only check if package is up to date"
@@ -719,6 +720,8 @@
 				PKGMK_RECURSIVE="yes" ;;
 			-d|--download)
 				PKGMK_DOWNLOAD="yes" ;;
+			-ic|--ignore-conflict)
+				PKGMK_IGNORE_CONFLICT="-f" ;;
 			-do|--download-only)
 				PKGMK_DOWNLOAD="yes"
 				PKGMK_DOWNLOAD_ONLY="yes" ;;
